### PR TITLE
Switching between Media Gallery bookmarks does not change the selected folder

### LINKED ISSUE #1523: Switching between Views does not change the selected folder. [Media Gallery]
### Preconditions (*)
1. Installed Magento with Adobe Stock Integration
2. Configured  integration in Stores -> Configuration -> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery is enabled

### Steps to reproduce (*)
1. Go **Content - Pages** click **Add New Page**
2. Expand **Content** click **Show / Hide Editor**, **Insert Image...**
3. Create a main folder named _folder_ for example and select it
main folder
4. Click on **Default View** dropdown 
5. Click **Save View As...**, and click **Save all changes** ( ➡️  )
_now you have a New View with the selected folder_
6. Close the Media Gallery and go to **Catalog - Categories**
7. Expand **Content** and click **Select from Gallery**
_folder is selected_
8. Try to switch between **New View** and **Default View**
9. Observe the _Active filters: Directory_

### Expected Result (*)
Catalog/category folder is selected, and the Default view is selected

### Actual Result (*)
The _Active filters: Directory_ is changed but the selected folder remains the same 
Peek 2020-06-24 15-29

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

This PR will implement the fix to change the selected folder when switching views in Media Gallery.

### Related Pull Requests

N/A
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/adobe-stock-integration/issues/1523

### Manual testing scenarios (*)

Apart from the mentioned steps please check directory navigation/operations in general

1. Go **Content** - Pages click **Add New Page**
2. Expand **Content** click **Show / Hide Editor**, **Insert Image**...
3. Create a main folder named _folder_ for example and select it
4. Click on **Default View** dropdown
5. Click **Save View As**..., and click **Save all changes** ( ➡️ )
 _now you have a New View with the selected folder_
7. Close the Media Gallery and go to **Catalog - Categories**
8. Expand Content and click Select from Gallery
_folder is selected_
10. Try to switch between **New View** and **Default View**
11. The folder selected for that view must be highlighted/selected,
 i.e.: If **New View** is selected, _folder_ should be selected,
**Catalog/category** folder is selected if the **Default view** is selected

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 1]
```diff
@@ -239,26 +239,30 @@ define([
...
                 isMediaBrowser = !_.isUndefined(window.MediabrowserUtility),
                 currentTreePath;
 
-            currentTreePath = this.isFiltersApplied(currentFilterPath) || !isMediaBrowser ? currentFilterPath :
-                Base64.idDecode(window.MediabrowserUtility.pathId);
-
-            if (this.folderExistsInTree(currentTreePath)) {
-                this.locateNode(currentTreePath);
+            if (_.isUndefined(currentFilterPath)) {
+                this.clearFiltersHandle();
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js:255]: the else condition can be omitted if the `return;' is added after this line

File: app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 2]
```diff
@@ -281,8 +285,8 @@ define([
          * @param {String} currentFilterPath
          */
         isFiltersApplied: function (currentFilterPath) {
-            return !_.isUndefined(currentFilterPath) && currentFilterPath !== '' &&
-                currentFilterPath !== 'wysiwyg' && currentFilterPath !== 'catalog/category';
+            return !_.isUndefined(currentFilterPath) && currentFilterPath !== ''
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: I think `catalog/category` comparison should also be removed (cc @Nazar65 )

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @joweecaquicla ! Please see my comments

Can you please cover this case with an MFTF test

Also please check the static tests https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/29411/a3de7a89d23fb2784c34d24facce7d29/Statics/allure-report-b2b/index.html#categories/0a4a9c930701952d758f01ac4e2bb450/6a07c842b5a5c7f8/

**Comment by joweecaquicla** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @joweecaquicla ! The implementation looks good! Can you please check the failed tests

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run Database Compare, Functional Tests CE, Functional Tests EE

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8008](https://jira.corp.magento.com/browse/ENGCOM-8008) has been created to process this Pull Request

File: app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js (hunk 2 above)
Comment (Nazar65) [reply app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: this comparisons needed to cover behavior with pres elected folders, for example when bookmark applied and we open media gallery from category, category folder should be presented only if the bookmarks not available

File: app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 3]
```diff
@@ -285,8 +286,7 @@ define([
          * @param {String} currentFilterPath
          */
         isFiltersApplied: function (currentFilterPath) {
-            return !_.isUndefined(currentFilterPath) && currentFilterPath !== ''
-                && currentFilterPath !== 'catalog/category';
```
Comment (Nazar65) [inline app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: I think this changes should be reverted

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Review by engcom-Lima — CHANGES_REQUESTED** [review]
:x: QA Failed
**Case 1**
**Steps to Reproduce:**
1. Go to Catalog -> Categories
2. Select any Category
3. Expand "Content" section
4. Click on the "Select from Gallery" button
5. Delete the "catalog" folder
6. Close the window
7. Click on the "Select from Gallery" button again 

**Expected Result:**  The "catalog/category" folder is created, expanded and selected
**Actual Result:** The "catalog/category" folder is created but not expanded and not selected
Peek 2020-08-14 13-20

**Case 2**
**Steps to Reproduce:**
1. Go to Catalog -> Categories
2. Select any Category
3. Expand "Content" section
4. Click on the "Select from Gallery" button
5. Select different from the "catalog" folder
6. Close the window
7. Click on the "Select from Gallery" button again 

**Expected Result:** The selected folder should not change
**Actual Result:** The selected folder is changed.
Peek 2020-08-14 13-35

**Case 3**
**Steps to Reproduce:**
1. Go to Catalog -> Categories
2. Select any Category
3. Expand "Content" section
4. Click on the "Select from Gallery" button
5. Select different from the "catalog" folder
6. Close the window
7. Click on the "Select from Gallery" button again 
**Expected Result:** The "Active Filter" applied to the previously selected directory 
**Actual Result:** The "Active Filter" applied to wrong directory
Screenshot_95

**Case 4**
**Steps to Reproduce:**
1. Go to Catalog -> Categories
2. Select any Category
3. Expand "Content" section
4. Click on the "Select from Gallery" button
5. Delete the "catalog" folder
6. Close the window
7. Click on the "Select from Gallery" button again 

**Expected Result:** The "Delete Folder" button is active 
**Actual Result:**  The "Delete Folder" button is graduated gray 
Screenshot_96

**Case 5**
**Steps to Reproduce:**
1. Go to Content -> Media Gallery
2. Click on the "Upload Image" button
3. Select any image 
4. Press "Enter" or click "Open" button 

**Expected Result:** Image uploaded successfully. No errors
**Actual Result:** Endless loading of the page after trying to add image in Standalone Media Gallery. 
Screenshot_93

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@engcom-Lima the implementation has been updated:
Case 1: Expected behaviour, if the filter was dropped/selected and media gallery was reopened on the same page - the filter state should be presereved
Case 2: Fixed
Case 3: Fixed
Case 4: Expected behaviour (same as case 1)
Case 5: Fixed

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run Static Tests

**Comment by sivaschenko** [comment]
@magento run Static Tests

**Comment by sivaschenko** [comment]
@magento run Static Tests

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8008](https://jira.corp.magento.com/browse/ENGCOM-8008) has been created to process this Pull Request

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

The folder is selected and deselected correctly while switching Views

Manual testing scenario - https://studio.cucumber.io/projects/131313/test-plan/folders/1054245/scenarios/5131805

**Pull request merged**